### ISSUE TITLE
Allow disabling `find_package(SDL2 REQUIRED)`

### ISSUE THREAD

**Issue by ericoporto** [sezero/SDL_sound#5]
SDL_sound works fine with Emscripten, but it can't be built with CMake due to

https://github.com/sezero/SDL_sound/blob/master/CMakeLists.txt#L32

If possible, allow to pass a flag to disable the find_package, because it will fail in Emscripten environment when building to Webassembly.

**Comment by sezero** [comment]
I don't know anything about Emscripten, but how shall we satisfy SDL2 
library calls without requiring and linking to SDL2?

CC'ing @icculus too, for his attention.

**Comment by icculus** [comment]
Same question here.

**Comment by ericoporto** [comment]
I was thinking on a parameter similar to NOT_REQUIRE_SDL2. Emscripten has a SDL2 port and it will fill SDL2_LIBRARIES variable when you use `-s USE_SDL=2`.

```CMake
if(NOT_REQUIRE_SDL2)
find_package(SDL2 REQUIRED)
endif()
```

For now the alternative is to run sed with cmake to remove the find_package line.

**Comment by sezero** [comment]
> I was thinking on a parameter similar to NOT_REQUIRE_SDL2. Emscripten has a SDL2 port and it will fill SDL2_LIBRARIES variable when you use `-s USE_SDL=2`.
> 
> ```cmake
> if(NOT DEFINED NOT_REQUIRE_SDL2)
> find_package(SDL2 REQUIRED)
> endif()
> ```
> 
> the problem is just that find_package won't play well with `-s USE_SDL=2` but I don't know the proper way to handle this.
> 
> For now the alternative is to run sed with cmake to remove the find_package line.

@icculus should decide.

**Comment by ericoporto** [comment]
I opened an additional issue on Emscripten about find_package working with it's own ports in case there's a way to make it work internally, but I think it will take a little while before getting this working

https://github.com/emscripten-core/emscripten/issues/14003

**Comment by sezero** [comment]
I'll keep this open for a while

**Comment by sezero** [comment]
Looking at the Emscripten issue you filed, curious: If you use your emcmake
with `-s USE_SDL=2` and if it does fill `SDL2_LIBRARIES` and `SDL2_INCLUDE_DIRS`
does something like the following work? If it really does, @icculus: is it OK?
```diff
diff --git a/CMakeLists.txt b/CMakeLists.txt
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -1,6 +1,9 @@
 # SDL_sound; An abstract sound format decoding API.
 #
 # Please see the file LICENSE.txt in the source's root directory.
+#
+# To specify a custom SDL2 installation, run cmake with command line arguments like:
+#   -DSDL2_INCLUDE_DIRS=/opt/SDL2/include -DSDL2_LIBRARIES=/opt/SDL2/lib/libSDL2.so
 
 cmake_minimum_required(VERSION 2.8.12)
 
@@ -29,7 +32,9 @@
     add_definitions(-xldscope=hidden)
 endif()
 
-find_package(SDL2 REQUIRED)
+if((NOT SDL2_INCLUDE_DIRS) OR (NOT SDL2_LIBRARIES))
+    find_package(SDL2 REQUIRED)
+endif()
 include_directories(${SDL2_INCLUDE_DIRS})
 if(WIN32)
     # -lmingw32: gcc adds it automatically.
```

**Comment by ericoporto** [comment]
This looks super sensible and that it should work, but it doesn't. 

I am reading the sources and it appears it doesn't fill these variables at all (I thought it did!), the libraries and headers pre-built by Emscripten go into a `sysroot/lib` and `sysroot/include` but I can't understand how things are magically found by CMake.

But I think I found a way to bypass, I noticed it generates an empty FindOpenGL.cmake to make things like find_package with OpenGL work, and same for OpenAL. So I will do a PR to provide an empty FindSDL2.cmake when `-s USE_SDL=2` is used.
